comp/core/workloadmeta/collectors/internal/containerd/image.go

```diff
@@ -253,7 +256,7 @@ func (c *collector) notifyEventForImage(ctx context.Context, namespace string, i
 	// Build initial workloadmeta.ContainerImageMetadata from manifest and image
 	manifest, err := images.Manifest(ctxWithNamespace, img.ContentStore(), img.Target(), img.Platform())
 	if err != nil {
-		return fmt.Errorf("error getting image manifest: %w", err)
+		return nil, fmt.Errorf("error getting image manifest: %w", err)
```

Comment:
If I understand the issue was that we where returning an incomplete image? I'm right?

Reply:
Yes, when agent starts, the first image it sends is always incomplete. The backend process (in my understanding) has a cache to check imageId. So once first image payload is accepted, the second image metadata for the same image will be ignored until the cache expires (in 15 mins). The complete image metadata will be updated after 15 mins.